src/eva/vision/models/networks/decoders/conv.py

```diff
@@ -0,0 +1,91 @@
+"""Convolutional based semantic segmentation decoder."""
+
+from typing import List, Tuple
+
+import torch
+from torch import nn
+from torch.nn import functional
+
+from eva.vision.models.networks.decoders import decoder
+
+
+class ConvDecoder(decoder.Decoder):
+    """Convolutional segmentation decoder."""
+
+    def __init__(self, layers: nn.Module) -> None:
+        """Initializes the convolutional based decoder head.
+
+        Here the input nn layers will be directly applied to the
+        features of shape (batch_size, hidden_size, num_patches_height,
+        num_patches_width), where num_patches is image_size / patch_size.
+        Note the num_patches is also known as grid_size.
+
+        Args:
+            layers: The convolutional layers to be used as the decoder head.
+        """
+        super().__init__()
+
+        self._layers = layers
```

Comment:
Should we apply some sanity checks here to make sure that the `layers` passed in are compatible with this decoder?

Reply:
I'm not sure if we can cover all the edge cases here 🤔  I think like this is ok since its documented
what do you think?